software/32.Zig/README.md

```diff
@@ -0,0 +1,267 @@
+
+📚 Documentation:
+
+ -[What is a palindrome ?](https://www.wikiwand.com/en/Palindrome)
+ -[New function](https://ziglang.org/documentation/master/#Functions)
+ -[The types in Zig](https://ziglang.org/documentation/master/#Primitive-Types)
+ -[Control_Structures](https://zig.guide/language-basics/while-loops)
+
+ ✔️ Validation:
+
+    Here's a main example :
+
+    pub fn main() void {
+    const stdout = std.io.getStdOut().writer();
+
+    const test_word1 = "madam";
+    const test_word2 = "hello";
+
+    const result1 = is_palindrome(test_word1);
+    const result2 = is_palindrome(test_word2);
+
+    stdout.print("{} is palindrome: {}\n", .{test_word1, result1}) catch {};
+    stdout.print("{} is palindrome: {}\n", .{test_word2, result2}) catch {};
+}
+
+ When you compile and run palindrome.zig, the output should be:
+
+    ```sh
+    madam is palindrome: true
+    ```
+    or
+
+    ```sh
+    hello is palindrome: false
+    ```
+
+## Step 3 - Fibonacci sequence
+
+📑 Description:
+
+For the third exercise, you need to create a function that generates and displays the Fibonacci sequence up to a specified number of elements.
+
+📌 Tasks:
+
+- Create a file `fibonacci.zig` for this new function.
+
+For the third exercise, create a function that takes one parameter:
+
+- A number `max` with type `i32` that represent the number of element to compute.
+
+- You must now display the sequence of Fibonacci from the number of starts to the `max` value.
+
+Here is a small example of the beginning of the Fibonacci sequence:
+
+    ```shell
+    0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, ...
+    ```
+
+📚 Documentation:
+
+ -[Fibonacci number](https://www.wikiwand.com/en/Fibonacci_number)
+ -[Match](https://ziglang.org/documentation/master/#Standard-Library-Math-Functions)
+ -[Vector](https://ziglang.org/documentation/master/#Vectors)
+
+  ✔️ Validation:
+
+    Given the Fibonacci sequence (see previous example), if max is 5, the function should output:
+
+    ```sh
+    0, 1, 1, 2, 3
+    ```
+
+## Step 4 - A simple and useful `Calculator`
+
+📑 Description:
+
+Let's go further now!
+
+The objective of this fifth exercise is to create a simplifying calculator.
+
+📌 Tasks:
+
+- Create a file `calculator.zig` for this new function.
+
+To do this, you have to make a calculator that can do:
+
+- `Addition`
+- `Multiplication`
+- `Division`
+- `Subtraction`
+
+- between two values that must be retrieved one after the other from the user's input.
+
+> 💡 Pay attention to your error handling !
+
+- Display the result of the calculation in your terminal.
+
+📚 Documentation:
+ -[Errors](https://ziglang.org/documentation/master/#Errors)
+ -[Memory](https://zig.guide/standard-library/allocators)
+ -[ArrayList](https://zig.guide/standard-library/arraylist)
+
+  ✔️ Validation:
+
+    You should get something like this :
+
+    ```sh
+    First number: 5
+    Operation: +
+    Second number: 3
+    8
+    ```
+
+    Here's a typical error handeling example :
+
+    ```sh
```

Comment:
Maybe the error message should occur instantly after he enters the information instead of waiting till he finishes.